plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java

```diff
@@ -124,7 +124,7 @@ public void onSuccess(StatusResponse result)
                             {
                                 verify(result != null);
 
-                                if (!(result.getStatusCode() >= 200 && result.getStatusCode() < 300) && attempt < config.getRetryCount()) {
+                                if (shouldRetry(result) && attempt < config.getRetryCount()) {
```

Comment:
i think we should split this into 2 ifs, so when `shouldRetry` is true but there are no retries left it is communicated

Reply:
this is something i should have implemented, sorry about that